NewsFeed/Common/ServerRequest.swift

```diff
@@ -15,33 +15,53 @@ class ServerRequest {
 // MARK: - Properties
     var newsArray: [News] = []
     var isLoading = false
+  
+    // Private
+    private var dataTask: URLSessionDataTask?
```

Comment:
I don't think you need this as a member/property. Looks like it can be used as a local variable in `func performRequest(...)`.